bindings/node/lib/clientEncryption.js

```diff
@@ -549,6 +554,75 @@ module.exports = function (modules) {
       return value;
     }
 
+    /**
+     * Creates a collection that has encrypted document fields
+     *
+     * @template {TSchema} - Schema for the collection being created
+     * @param {Db} db - A database to create the collection in
+     * @param {string} name - The name of the collection to be created
+     * @param {object} options - Options for createDataKey and for createCollection.
+     * @param {string} options.provider - provider name
+     * @param {ClientEncryptionCreateDataKeyProviderOptions} options.createDataKeyOptions - options to pass to createDataKey
+     * @param {CreateCollectionOptions} options.createCollectionOptions - options to pass to createCollection, must include `encryptedFields`
+     * @returns {Promise<{ collection: Collection<TSchema>, encryptedFields: Document }>} - created collection and generated encryptedFields
+     * @throws {MongoCryptCreateEncryptedCollectionError} - If part way through the process createDataKey fails, an error will be rejected that has the `encryptedFields` that were created.
```

Comment:
As I understand the code, it’s not just the `encrypedFields` that were created, it’s also including the fields that were already present before (potentially with a non-null `keyId`)?

Reply:
That's correct, based on these bits of the spec:

- If F (a field from fields) is not a document element, skip it.
- Otherwise, if F has a "keyId" named element K and K is a null value:
  - Let D be the result of CE.createDataKey(kmsProvider, dkOpts).
  - etc...

Reply:
Followed up https://github.com/mongodb/specifications/pull/1375 we should throw an error that still communicates the `encryptedFields` settings. We now have two distinct error classes